projects/vdk-core/src/vdk/internal/builtin_plugins/job_properties/base_properties_impl.py

```diff
@@ -1,13 +1,14 @@
 # Copyright 2021-2023 VMware, Inc.
 # SPDX-License-Identifier: Apache-2.0
-from typing import List
-from typing import Type
 
 
-def check_valid_property(k: str, v: str, supported_types: List[Type] = []) -> None:
+def check_valid_property(k: str, v: str, supported_types=None) -> None:
     """
     Check if property key and value are valid
     """
+    if supported_types is None:
+        supported_types = []
+
     if str != type(k) or k.strip() != k or "".join(k.split()) != k:
         msg = (
             f"Property {k} is of unsupported type or has unsupported name. "
```

Comment:
there is a bug on line 24 of this file. it always says f"Only int, float, str, list, and NoneType types are supported. ". 
Even if they are not the types in the supportes_types list.

Reply:
Codacy check suggestion